### ISSUE TITLE
jacoco test coverage broken in 2.9.X

### ISSUE THREAD

**Issue by eformat** [quarkusio/quarkus#25673]
### Describe the bug

we have reliably been using the external jacoco config for code coverage with quakus for quite some time

https://quarkus.io/guides/tests-with-coverage#coverage-for-tests-not-using-quarkustest

the code coverage results are nill using the latest 2.9.X - quarkus-jacoco version
works fine reverting to 2.8.3.Final (and prior versions) 

### Expected behavior

working code coverage `mvn test` using 2.8.3.Final

![Screenshot from 2022-05-19 12-59-31](https://user-images.githubusercontent.com/712608/169194980-a5863c48-c816-4907-bbe3-33ced09928c8.png)

### Actual behavior

not working code coverage `mvn test` using 2.9.1.Final

![Screenshot from 2022-05-19 12-56-05](https://user-images.githubusercontent.com/712608/169195062-e210183e-34d8-44a0-a767-db052abc2352.png)

### How to Reproduce?

i need to narrow this down, but for now ...
```
git clone https://github.com/petbattle/pet-battle-api
# update pom.xml - https://github.com/petbattle/pet-battle-api/blob/master/pom.xml#L17
# to use 2.9.1.Final  (also breaks in 2.9.0)
mvn test
```

### Output of `uname -a` or `ver`

Linux virt 5.17.6-300.fc36.x86_64 #1 SMP PREEMPT Mon May 9 15:47:11 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19 OpenJDK Runtime Environment 21.9 (build 17.0.3+7) OpenJDK 64-Bit Server VM 21.9 (build 17.0.3+7, mixed mode, sharing)

### Quarkus version or git rev

2.9.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d) Maven home: /opt/apache-maven-3.8.1 Java version: 17.0.3, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-17-openjdk-17.0.3.0.7-1.fc36.x86_64 Default locale: en_NZ, platform encoding: UTF-8 OS name: "linux", version: "5.17.6-300.fc36.x86_64", arch: "amd64", family: "unix"

### Additional information

We have been using the jacoco 0.8.8 which i know is pretty new - but it contains fixes that make this work !

**Comment by geoand** [comment]
It's really hard to figure out exactly what's going on because the setup in the project is complex, but this does not appear to be a Quarkus issue.
Changing the jacoco version in the project to `0.8.7` seems to get the coverage report back.
Furthermore, our coverage quickstart works just fine, so I am going to close this.

**Comment by eformat** [comment]
thanks @geoand for looking at this !
i am going to debug whats going on - i had not thought to try the downgrade since 0.8.7 was broken with earlier versions, so had been using 0.8.8-SNAPSHOT for a long time ! ha ! 

FWIW i had been tracking a bug in similar area (quarkus, jacoco, offline)
https://github.com/quarkusio/quarkus/issues/9837

which was finally fixed and merged in 0.8.8 jacoco - https://github.com/jacoco/jacoco/issues/331

it took over a year between 0.8.7 and 0.8.8

so, perversely .. i now have the situation where:

2.8.3.Final - works with 0.8.8 jacoco but NOT 0.8.7
2.9.1.Final - works with 0.8.7 jacoco but NOT 0.8.8

let me find a simpler reproducer ...

**Comment by eformat** [comment]
I have found a workaround for my case, i will leave this here for anyone who needs it.

Basically the configuration for testing when not using QuarkusTest in the pom.xml was only needed in my case because of the use of allure-junit5. We are using this to report and attach test output into Allure https://github.com/allure-framework

The default allure aspect4j/annotation mechanism (@Attachment) returns the results as attachments. Removing this and calling the framework methods instead allows us to **NOT** have to use the quarkus settings described here (these were used previously):

https://quarkus.io/guides/tests-with-coverage#coverage-for-tests-not-using-quarkustest

Coverage works fine using latest versions of dependencies and plugins (quarkus 2.9.1.Final, jacoco 0.8.8, alllure 2.18.0)

Given the 0.8.7 jacoco maven-plugin works with the quarkus-2.9.1 (which has the latest jacoco 0.8.8 as a dep) it is very likely that aspect4j/allure is causing the JVM to exit early - such that the results are not written. On debugging into jacoco, the coverage data was calculated ok, just not written to the report.
